pkg/store/etcd.go

```diff
@@ -176,48 +175,39 @@ func (s *Etcd) CancelWatch(key string) error {
 
 // AtomicPut put a value at "key" if the key has not been
 // modified in the meantime, throws an error if this is the case
-func (s *Etcd) AtomicPut(key string, oldValue []byte, newValue []byte, index uint64) (bool, error) {
-	resp, err := s.client.CompareAndSwap(normalize(key), string(newValue), 5, string(oldValue), 0)
+func (s *Etcd) AtomicPut(key string, value []byte, previous *KVPair) (bool, error) {
```

Comment:
@abronan Need your review of this. I think atomic operations were broken in etcd